guardian/media-atom-maker
# Modernise media-atom-maker's dependencies

## What does this change?

This PR has (at time of writing) 26 open PRs, almost entirely due to security dependencies from dependabot. This appears to primarily be a symptom of the app using very old versions of webpack, react, eslint and jest among others.

Bring media-atom-maker to Webpack v4, and upgrade any linked dependencies that also need upgrading. This will allow us to patch security vulnerabilities more easily in the future as well as develop on the app since a large number of the libraries are currently unsupported and deprecated.

This PR bumps a number of core dependencies to a modern (or even latest) version, allowing us to make use of current docs and features in the future. Namely, the bumped dependencies are:

* Webpack & Webpack loaders (css/sass-loader)
* React & React-dom
* Eslint & Prettier

These upgrades introduced breaking changes that needed to be addressed, mainly in the form of:

* Updating `webpack.{env}.conf.js` config files to the new syntax
* Making the required changes to the library implementation (eg `react-datepicker` using `Date` types as opposed to `moment` and then prettifying the file, causing a larger diff
* Using relative imports for `url()` files in scss, as per new webpack rules

## Definition of Success

The app works exactly like before, maybe even better!

## How to test

Clone the repository locally, run the setup and start scripts and check that it works as intended. Alternatively, check CODE.
